statsmodels/genmod/generalized_linear_model.py

```diff
@@ -1129,25 +1176,28 @@ def null(self):
         if len(kwargs) > 0:
             return GLM(endog, exog, family=self.family, **kwargs).fit().mu
         else:
-            wls_model = lm.WLS(endog, exog, weights=self._data_weights)
+            wls_model = lm.WLS(endog, exog,
+                               weights=self._freq_weights * self._n_trials)
             return wls_model.fit().fittedvalues
 
     @cache_readonly
     def deviance(self):
-        return self.family.deviance(self._endog, self.mu)
+        return self.family.deviance(self._endog, self.mu, self._freq_weights)
```

Comment:
in general use keywords for keyword arguments.

this PR breaks backwards compatibility if keyword arguments are used as positional arguments.